include/zephyr/net/prometheus/pm_collector.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief Prometheus collector APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_PM_COLLECTOR_H_
```

Comment:
The header files don't need to be namespaced if the namespace is in the folder name:
```
#include <zephyr/net/prometheus/pm_collector.h>
```
can be shortened to
```
#include <zephyr/net/prometheus/collector.h>
```
as `pm` / `prometheus` is already in the path :)

Reply:
The include guard should also be updated to use the namespace `PROMETHEUS` instead of `PM`
```
#ifndef ZEPHYR_INCLUDE_PROMETHEUS_COLLECTOR_H_
```

Reply:
Now we'r talking :). I did change right now

Reply:
Nice!